traefik.go

```diff
@@ -209,29 +209,36 @@ func main() {
 	log.Info("Shutting down")
 }
 
-func createTLSConfig(certFile string, keyFile string) (*tls.Config, error) {
+// creates a TLS config that allows terminating HTTPS for multiple domains using SNI
+func createTLSConfig(certs []Certificate) (*tls.Config, error) {
 	config := &tls.Config{}
 	if config.NextProtos == nil {
 		config.NextProtos = []string{"http/1.1"}
 	}
 
 	var err error
-	config.Certificates = make([]tls.Certificate, 1)
-	if len(certFile) > 0 && len(keyFile) > 0 {
-		config.Certificates[0], err = tls.LoadX509KeyPair(certFile, keyFile)
-		if err != nil {
-			return nil, err
+	if len(certs) > 0 {
+		config.Certificates = make([]tls.Certificate, len(certs))
+		for i, v := range certs {
+			config.Certificates[i], err = tls.LoadX509KeyPair(v.CertFile, v.KeyFile)
+			if err != nil {
+				return nil, err
+			}
 		}
+		// BuildNameToCertificate parses the CommonName and SubjectAlternateName fields
+		// in each certificate and populates the config.NameToCertificate map.
+		config.BuildNameToCertificate()
 	} else {
 		return nil, nil
 	}
+
 	return config, nil
 }
 
 func startServer(srv *manners.GracefulServer, globalConfiguration *GlobalConfiguration) {
 	log.Info("Starting server")
-	if len(globalConfiguration.CertFile) > 0 && len(globalConfiguration.KeyFile) > 0 {
-		err := srv.ListenAndServeTLS(globalConfiguration.CertFile, globalConfiguration.KeyFile)
+	if len(globalConfiguration.Certificates) > 0 {
+		err := srv.ListenAndServeTLS("", "")
```

Comment:
Not shure about that. If certs is not empty, you should build a TLSConfig and pass it to the server no?